### PR TITLE
adding vectio files

### PR DESCRIPTION
**Pull request by sakshamagarwals**

### CODE REVIEW

File: homatransport/src/application/WorkloadSynthesizer.cc [hunk 1]
```diff
@@ -16,6 +16,7 @@
 #include <unordered_set>
```
Comment (behnamm) [inline homatransport/src/application/WorkloadSynthesizer.cc:16]: Not for this commit, but later remove "homatransport/" parent folder. Let's have everything based on "vectio" naming.

File: homatransport/src/application/WorkloadSynthesizer.cc [hunk 2]
```diff
@@ -143,6 +146,11 @@ WorkloadSynthesizer::initialize()
     startTime = par("startTime").doubleValue();
     stopTime = par("stopTime").doubleValue();
     xmlConfig = par("appConfig").xmlValue();
+    std::string OutputFileName = std::string(
+                "results/") + std::string(par("resultFile").stringValue());
+    if(!outputFile.is_open()) {
```
Comment (behnamm) [inline homatransport/src/application/WorkloadSynthesizer.cc:152]: Why is this needed? Doesn't the omnet++ infrastructure open/close the result file itself?

File: homatransport/src/application/WorkloadSynthesizer.cc [hunk 3]
```diff
@@ -143,6 +146,11 @@ WorkloadSynthesizer::initialize()
     startTime = par("startTime").doubleValue();
     stopTime = par("stopTime").doubleValue();
     xmlConfig = par("appConfig").xmlValue();
+    std::string OutputFileName = std::string(
+                "results/") + std::string(par("resultFile").stringValue());
```
Comment (behnamm) [inline homatransport/src/application/WorkloadSynthesizer.cc]: make "restults/" a const char*

File: homatransport/src/application/WorkloadSynthesizer.cc [hunk 4]
```diff
@@ -534,6 +542,16 @@ WorkloadSynthesizer::processRcvdMsg(cPacket* msg)
     mesgStats.transportSchedDelay =  rcvdMsg->getTransportSchedDelay();
     emit(mesgStatsSignal, &mesgStats);
 
+    inet::L3Address srcAddr = rcvdMsg->getSrcAddr();
+    inet::L3Address destAddr = rcvdMsg->getDestAddr();
+
+    int srcId = srcAddr.toIPv4().getDByte(2) * 16 + srcAddr.toIPv4().getDByte(3);
```
Comment (behnamm) [inline homatransport/src/application/WorkloadSynthesizer.cc]: Add a comment how this id calculation is working

File: homatransport/src/transport/VectioTransport.h [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+ * message not yet send. This transport will only be near optimal if the network
+ * has priority queues and only for many-senders/single-receiver scenario. The
+ * scheduling mechanism for the priority queues in the network would always
+ * choose the lowest priority packet that belongs to shortes message among all
+ * packets in the queue.
+ */
+
+class VectioTransport : public cSimpleModule
+{
+  PUBLIC:
```
Comment (behnamm) [inline homatransport/src/transport/VectioTransport.h]: Let's replace these capital case PUBLIC/PRIVATE/etc. with lower cases. I originally had them for testing purpose.

File: homatransport/src/application/WorkloadSynthesizer.cc [hunk 6]
```diff
@@ -30,6 +31,8 @@ simsignal_t WorkloadSynthesizer::msgE2EDelaySignal =
     registerSignal("msgE2EDelay");
 simsignal_t WorkloadSynthesizer::mesgStatsSignal = registerSignal("mesgStats");
 
+std::ofstream outputFile;
+
 WorkloadSynthesizer::WorkloadSynthesizer()
 {
     msgSizeGenerator = NULL;
```
Comment (behnamm) [inline homatransport/src/application/WorkloadSynthesizer.cc:38]: Would it be possible to move to Google naming style for variables and functions? I sent the style guide to you in chat

**Review by behnamm — CHANGES_REQUESTED** [review]

**Review by hwassel — COMMENTED** [review]
Can we format the C++ (and headers) to follow the 80-char limit?

File: homatransport/src/application/WorkloadSynthesizer.cc (hunk 1 above)
Comment (sakshamagarwals) [reply homatransport/src/application/WorkloadSynthesizer.cc:16]: Will take care of it in the next PR

File: homatransport/src/application/WorkloadSynthesizer.cc (hunk 2 above)
Comment (sakshamagarwals) [reply homatransport/src/application/WorkloadSynthesizer.cc:152]: Using my own logfiles over the default logging in omnet right now.

File: homatransport/src/application/WorkloadSynthesizer.cc (hunk 4 above)
Comment (sakshamagarwals) [reply homatransport/src/application/WorkloadSynthesizer.cc]: Actually this is not working as intended right now due to ipv4configurator assigning random addresses, will get rid of this for now.

File: homatransport/src/application/WorkloadSynthesizer.cc (hunk 6 above)
Comment (sakshamagarwals) [reply homatransport/src/application/WorkloadSynthesizer.cc:38]: Like we discussed, for this project using camelCase for this project.

**Review by behnamm — APPROVED** [review]
Please handle the remaining comments and push this commit.

**Review by behnamm — APPROVED** [review]

**Review by behnamm — APPROVED** [review]
